packages/core/types/src/modules-sdk/index.ts

```diff
@@ -87,6 +87,8 @@ export type ModuleDefinition = {
   isRequired?: boolean
   isQueryable?: boolean // If the module is queryable via Remote Joiner
   dependencies?: string[]
+  // Only to be used in exceptional cases - relying on the shared container will break the isolation of the module
+  passSharedContainer?: boolean
```

Comment:
I would suggest using a name that seems more like an "internal" prop. eg: `__passAppContainer` and comment it with `/** @internal */`

Reply:
LGTM

would be ideal to start listening to jobs `onApplicationStart`

Reply:
Sounds good 👍